app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,35 @@
+class SessionsController < ApplicationController
+  def login
+    @user = User.find_by(name: params[:name])
+    if @user
+      flash[:success] = "Logged in as #{@user.name}"
+      session[:user_id] = @user.id
+      session[:user] = @user.name
+      redirect_to root_path
+    else
+      @user = User.new(name: params[:name])
```

Comment:
this line says "take whatever is stored in `params[:name]` and then put it into the key `name` for a new `User`.

In the log_form, params gets the value from the text field and places it into `params[:username]`, so at this point in the controller, `params[:name]` is nil and the user doesn't save.